app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
+
+  def saml_response_url
+    saml_request.response_url
+  end
```

Comment:
Gotcha. You don't think other applications would benefit from this method? Would it hurt to propose it upstream and see what they say?

Also, `helper_method` in Rails is used to make the method available to a view, but in this case, we are not calling `saml_response_url` in any views, so the `helper_method` call is not necessary.